packages/domain-picker/src/domain-picker/use-your-domain-item.tsx

```diff
@@ -37,7 +47,7 @@ const UseYourDomainItem: React.FunctionComponent< Props > = ( { onClick } ) => {
 					</span>
 				</div>
 			</div>
-			<ArrowButton arrow="right" onClick={ onClick }>
+			<ArrowButton arrow="right" onClick={ onClickWithNoPropagation }>
```

Comment:
Is there a reason for not just removing the `onClick` prop here? When the `label` was replaced with `WrappingElement` in https://github.com/Automattic/wp-calypso/pull/49249 and the extra `onClick` was added, this one started to be redundant IMO 🤔 
cc: @StefanNieuwenhuis

Reply:
Good point. I guess I was being too conservative in my approach — I just applied you suggestion instead in https://github.com/Automattic/wp-calypso/pull/51074/commits/57accf97b8298775fa26e56e6743af9b9497ab17 and it seems to work well. Thank you!